optaplanner-core/src/main/java/org/optaplanner/core/impl/score/buildin/bendable/BendableScoreHolderImpl.java

```diff
@@ -74,9 +73,9 @@ public int getSoftScore(int softLevel) {
     @Override
     public void configureConstraintWeight(Rule rule, BendableScore constraintWeight) {
         super.configureConstraintWeight(rule, constraintWeight);
-        BiConsumer<RuleContext, Integer> matchExecutor;
+        IntMatchExecutor matchExecutor;
```

Comment:
This (and similar changes in other `ScoreHolder`s) removes *a lot* of auto-boxing.